Formula/corsixth.rb

```diff
@@ -54,24 +58,53 @@ def install
       end
     end
 
+    datadir = OS.mac? ? prefix/"CorsixTH.app/Contents/Resources/" : pkgshare
     system "cmake", ".", "-DLUA_INCLUDE_DIR=#{lua.opt_include}/lua",
-                         "-DLUA_LIBRARY=#{lua.opt_lib}/liblua.dylib",
+                         "-DLUA_LIBRARY=#{lua.opt_lib/shared_library("liblua")}",
                          "-DLUA_PROGRAM_PATH=#{lua.opt_bin}/lua",
-                         "-DCORSIX_TH_DATADIR=#{prefix}/CorsixTH.app/Contents/Resources/",
+                         "-DCORSIX_TH_DATADIR=#{datadir}",
                          *std_cmake_args
     system "make"
-    cp_r %w[CorsixTH/CorsixTH.lua CorsixTH/Lua CorsixTH/Levels CorsixTH/Campaigns CorsixTH/Graphics CorsixTH/Bitmap],
-         "CorsixTH/CorsixTH.app/Contents/Resources/"
-    prefix.install "CorsixTH/CorsixTH.app"
+    if OS.mac?
+      resources = %w[
+        CorsixTH/CorsixTH.lua
+        CorsixTH/Lua
+        CorsixTH/Levels
+        CorsixTH/Campaigns
+        CorsixTH/Graphics
+        CorsixTH/Bitmap
+      ]
+      cp_r resources, "CorsixTH/CorsixTH.app/Contents/Resources/"
+      prefix.install "CorsixTH/CorsixTH.app"
+    else
+      system "make", "install"
+    end
 
     lua_env = { LUA_PATH: ENV["LUA_PATH"], LUA_CPATH: ENV["LUA_CPATH"] }
-    (bin/"CorsixTH").write_env_script(prefix/"CorsixTH.app/Contents/MacOS/CorsixTH", lua_env)
+    bin_path = OS.mac? ? prefix/"CorsixTH.app/Contents/MacOS/CorsixTH" : bin/"corsix-th"
+    (bin/"CorsixTH").write_env_script(bin_path, lua_env)
```

Comment:
Do we want to have both `corsix-th` (without env) and `CorsixTH` (with env) symlinked on Linux?

Reply:
Probably not, but I'm not entirely sure how to hide `corsix-th` on Linux without breaking something.  While the macOS installation convenient buries the original binary in a folder that is not linked, on Linux it installs directly to `bin`.  If it's relocatable I suppose it could be moved to `libexec`, but otherwise I'm not really sure how to hide it.

Reply:
I guess that is fine. In current state of upstream, I think it is possible to install binaries into `libexec` or `libexec/bin` (e.g. via `CMAKE_INSTALL_BINDIR`) as there isn't any `lib` and non-relocatable part is from `share`, but it is possible this won't remain the case and it is difficult to test functionality of GUIs.

Reply:
I changed this so that `CMAKE_INSTALL_BINDIR` goes to `libexec/bin` on Linux and changed the `env_script` accordingly, so now we only link the `CorsixTH` `env_scrpt` into `bin`.  I also added `mesa` as a dependency, so I think this is ready now.